### PR TITLE
Renames Digital Ocean cloud _facts modules to _info

### LINKED ISSUE #57276: Digital Ocean cloud _facts modules need to be renamed to _info
##### SUMMARY
Modules should only be called `_facts` if they return `ansible_facts`, and `ansible_facts` should only be used for system-specific information. Information on cloud providers is not system specific. (See #54280 and [this checklist](https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_checklist.html#contributing-to-ansible-objective-requirements).) Since this has only been clarified shortly before Ansible 2.8 went into feature freeze, it was decided to do bulk renames for Ansible 2.9.

Below you can find a list of modules from [modules/cloud/digital_ocean](https://github.com/ansible/ansible/tree/devel/lib/ansible/modules/cloud/digital_ocean/) which are called `_facts` but seem to not return `ansible_facts` at all. These should be renamed (similarly to #56822).

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
digital_ocean_account_facts
digital_ocean_certificate_facts
digital_ocean_domain_facts
digital_ocean_firewall_facts
digital_ocean_floating_ip_facts
digital_ocean_image_facts
digital_ocean_load_balancer_facts
digital_ocean_region_facts
digital_ocean_size_facts
digital_ocean_snapshot_facts
digital_ocean_tag_facts
digital_ocean_volume_facts

##### ANSIBLE VERSION
2.9.0

### PR DESCRIPTION
**Pull request by pratikgadiya12**
##### SUMMARY
Renames Digital Ocean _facts to _info since they do not return ansible_facts.
Continues #56822
Fixes #57276

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
digital_ocean_account_facts
digital_ocean_certificate_facts
digital_ocean_domain_facts
digital_ocean_firewall_facts
digital_ocean_floating_ip_facts
digital_ocean_image_facts
digital_ocean_load_balancer_facts
digital_ocean_region_facts
digital_ocean_size_facts
digital_ocean_snapshot_facts
digital_ocean_tag_facts
digital_ocean_volume_facts

### CODE REVIEW

**Comment by Akasurde** [comment]
@pratikgadiya12 Please check this error - 

```
01:13 ERROR: Found 1 rstcheck issue(s) which need to be resolved:
01:13 ERROR: docs/docsite/rst/porting_guides/porting_guide_2.9.rst:79:0: Bullet list ends without a blank line; unexpected unindent. (100%)
```

File: docs/docsite/rst/porting_guides/porting_guide_2.9.rst [hunk 1]
```diff
@@ -76,6 +76,18 @@ Noteworthy module changes
 * The ``bigip_device_facts`` module was renamed to :ref:`bigip_device_info <bigip_device_info_module>`.
 * The ``bigiq_device_facts`` module was renamed to :ref:`bigiq_device_info <bigiq_device_info_module>`.
 * The ``one_image_facts`` module was renamed to :ref:`one_image_info <one_image_info_module>`.
+- The ``digital_ocean_account_facts`` module was renamed to :ref:`digital_ocean_account_info <digital_ocean_account_info_module>`.
```
Comment (felixfontein) [inline docs/docsite/rst/porting_guides/porting_guide_2.9.rst]: Use `*` instead of `-` to continue this list.

**Review by felixfontein — COMMENTED** [review]
The files starting with an underscore - are these symlinks to the corresponding `_info` files? It currently doesn't look like that.

Also, you should include snippts like [this](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/system/python_requirements_info.py#L120-L121) in the module code so that users will be informed that the name changed. Finally, including a note like [this](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/system/python_requirements_info.py#L19) in the module docs would be helpful as well.

File: docs/docsite/rst/porting_guides/porting_guide_2.9.rst (hunk 1 above)
Comment (Akasurde) [reply docs/docsite/rst/porting_guides/porting_guide_2.9.rst]: Nice catch.

Comment (pratikgadiya12) [reply docs/docsite/rst/porting_guides/porting_guide_2.9.rst]: fixed

**Comment by pratikgadiya12** [comment]
@felixfontein @Akasurde Thanks for the review. Supplied comments are fixed.

**Review by pmarques — APPROVED** [review]
LGTM

File: lib/ansible/modules/cloud/digital_ocean/_digital_ocean.py [hunk 2]
```diff
@@ -1,487 +0,0 @@
-#!/usr/bin/python
```
Comment (felixfontein) [inline lib/ansible/modules/cloud/digital_ocean/_digital_ocean.py]: Why is this removed? It is not supposed to be removed until 2.12.

**Review by felixfontein — CHANGES_REQUESTED** [review]
The renames look fine, but I don't think `_digital_ocean.py` should be removed.

File: lib/ansible/modules/cloud/digital_ocean/_digital_ocean.py (hunk 2 above)
Comment (pmarques) [reply lib/ansible/modules/cloud/digital_ocean/_digital_ocean.py]: True, missed this one

Comment (pratikgadiya12) [reply lib/ansible/modules/cloud/digital_ocean/_digital_ocean.py]: I missed that.

Comment (pratikgadiya12) [reply lib/ansible/modules/cloud/digital_ocean/_digital_ocean.py]: Fixed

**Review by felixfontein — APPROVED** [review]
The only difference to before was that the PR was apparently rebased to current devel, and that `_digital_ocean.py` is now included again. So now it looks good to me!

shipit

**Review by pmarques — APPROVED** [review]

**Comment by pmarques** [comment]
shipit

File: docs/docsite/rst/porting_guides/porting_guide_2.9.rst [hunk 3]
```diff
@@ -78,6 +78,18 @@ Noteworthy module changes
...
+* The ``digital_ocean_domain_facts`` module was renamed to :ref:`digital_ocean_domain_info <digital_ocean_domain_info_module>`.
+* The ``digital_ocean_firewall_facts`` module was renamed to :ref:`digital_ocean_firewall_info <digital_ocean_firewall_info_module>`.
+* The ``digital_ocean_floating_ip_facts`` module was renamed to :ref:`digital_ocean_floating_ip_info <digital_ocean_floating_ip_info_module>`.
+* The ``digital_ocean_image_facts`` module was renamed to :ref:`digital_ocean_image_info <digital_ocean_image_info_module>`.
+* The ``digital_ocean_load_balancer_facts`` module was renamed to :ref:`digital_ocean_load_balancer_info <digital_ocean_load_balancer_info_module>`.
+* The ``digital_ocean_region_facts`` module was renamed to :ref:`digital_ocean_region_info <digital_ocean_region_info_module>`.
+* The ``digital_ocean_size_facts`` module was renamed to :ref:`digital_ocean_size_info <digital_ocean_size_info_module>`.
+* The ``digital_ocean_snapshot_facts`` module was renamed to :ref `digital_ocean_snapshot_info <digital_ocean_snapshot_info_module>`.
```
Comment (samccann) [inline docs/docsite/rst/porting_guides/porting_guide_2.9.rst]:
```suggestion
* The ``digital_ocean_snapshot_facts`` module was renamed to :ref:`digital_ocean_snapshot_info <digital_ocean_snapshot_info_module>`.
```

**Review by samccann — CHANGES_REQUESTED** [review]

File: docs/docsite/rst/porting_guides/porting_guide_2.9.rst (hunk 3 above)
Comment (samccann) [reply docs/docsite/rst/porting_guides/porting_guide_2.9.rst]: minor fix to correct a broken link.

Comment (pratikgadiya12) [reply docs/docsite/rst/porting_guides/porting_guide_2.9.rst]: Good catch

Comment (pratikgadiya12) [reply docs/docsite/rst/porting_guides/porting_guide_2.9.rst]: fixed

**Review by felixfontein — APPROVED** [review]
shipit

**Review by samccann — APPROVED** [review]

**Comment by felixfontein** [comment]
@pratikgadiya12 thank you very much for doing the rename!
@Akasurde @pmarques @samccann thanks a lot for reviewing this!
